src/public/js/util.js

```diff
@@ -286,7 +293,7 @@ var UTIL = (function() {
     for (var i = 0; i < polynomial.length; i++) {
       scaledPolynomial[i] = fieldMult(polynomial[i], scalar);
     }
-    return scaledPolynomial;
+    return trimLeadingZeros(scaledPolynomial);
```

Comment:
done :+1: caught the scalar == 0 case at the beginning so we wouldn't get an array of lots of zeros